powerlevel9k.zsh-theme

```diff
@@ -1505,6 +1517,22 @@ prompt_todo() {
   fi
 }
 
+# taskwarrior: show data from taskwarrior
+prompt_tw() {
+  if $(hash task 2>&-); then
```

Comment:
You are calling `task` four times in your segment. Calling external programs is usually quite slow. Is there a way to fire one command, gathering all necessary data and searching in that data afterwards?

Reply:
I understand that this can be a problem but reading through
the TaskWarrior docs I could not find a way to do this. They
have an export function to return data in a JSON. One approach
would be to use `task +PENDING export` and count tasks based
on the "due" value but I don't think this would be faster.
What do you think ?

Reply:
You can get all the data from a single call, but that requires that you process the JSON data. You can do that as follows:
```zsh
    local -a tasks; tasks=( "${(@Qf)$( task export 2> /dev/null )}" ) # put the JSON file from task into an array, removing quotes
    local date=$( date +"%Y%m%d" 2> /dev/null )
    local today=0 over=0 pending=0
    local -a match
    for line in "${(s:,:)tasks[@]}"; do # split the array into multiple lines on delimiter comma (,)
      match=( ${(s.:.)line} ) # split line into keys and values
      [[ ${match[1]} == "status" && ${match[2]} == "pending" ]] && (( ++pending ))
      if [[ ${match[1]} == "due" ]]; then
        [[ ${match[2]:0:8} == $date ]] && (( ++today )) # tasks due today
        (( ${match[2]:0:8} < $date )) && (( ++over )) # overdue tasks
      fi
    done
```
This has the added advantage that you can get other information, like "priority" = "H" for how many high priority, etc.

Reply:
I don't know if it is faster though, because the JSON data exports all the tasks, including done and deleted, so the array can get massive.